sc-multisig-specs.md

```diff
@@ -0,0 +1,80 @@
+
+* **Proposer** - The proposer is an address whitelisted in the MSC that can propose any action. An action can be any transaction; for example: send 10 eGLD to the treasury, mint more ESDT, etc. All board members are proposers by default but non-board members can be added as well to the list of whitelisted proposers. The proposers can only propose actions that then need to be approved and signed by the board members. The board member that proposes an action doesn’t need to sign it anymore; it is considered signed.
+
+## Functionality
+The MSC should be able to perform most tasks that a regular account is able to perform. It should also be as general as possible. This means that it should operate with a generic concept of “Action”, that the board needs to sign before being performed. Actions can interact with the MSC itself (let's call them **internal actions**) or with external addresses or other SC (**external actions**).
+
+External actions have one and only one function, which is to send the action as a transaction whose sender is the MSC. Because any arbitrarily complex function call can be executed, the MSC functions exactly as a standard wallet, but requires multiple signatures.
+
+The types of internal actions should be the following:
+
+* Add a new member to the board.
+* Remove a member from the board. This is only allowed if the new board size remains larger than the number of required signatures (quorum). Otherwise a new member needs to be added first.
+* Change the quorum: the required number of signatures. Restriction: 1 <= quorum <= board size.
+* Add a proposer.
+* Remove a proposer.
+* Change multisig contract owner (might be relevant for upgrading the MSC).
+* Pay functions - by default we recommend the MSC to not be set up as payable and any deposit or send transaction of eGLD or ESDT towards the MSC will need to call the desired pay function (if a transaction is not a call to these 2 functions then it is rejected immediately and the value is sent back to original sender): Deposit and/or Send
+
+Any external and internal action will follow these steps and process:
+
+* **Propose action:** this will generate an action id. The action id is unique.
+* **View action:** the board members need to see the action proposed before they approve it.
+* **Sign action:** board members are allowed to sign. We might add an expiration date until board members can sign (until block x…).
+* **Un-sign action:** board members are allowed to un-sign, i.e. to remove their signature from an action. Actions with 0 signatures are cleared from storage. This is to allow mistakes to be cleared.
+* **Perform action (by id/hash)** - can be activated by proposers or board members. It is successful only if enough signatures are present from the board members. Whoever calls “perform action” needs to provide any eGLD required by the target, as well as to pay for gas.
```

Comment:
I assume that the value transferred and the transaction fee is taken from the MSC account. If this is correct, we might state this.

Reply:
Nope. As mentioned already: Whoever calls “perform action” needs to provide any eGLD required by the target, as well as to pay for gas. If there is a move balance kind of action, who call the action pays the gas and the amount to be moved is taken from MSC balance. But the gas is always taken from the balance on the one who create the "perform action" transaction. Added more details for clarity.